server/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -108,34 +109,36 @@ private void waitForEventsAndExecuteHealth(
     ) {
         assert request.waitForEvents() != null;
         if (request.local()) {
-            clusterService.submitStateUpdateTask(
-                "cluster_health (wait_for_events [" + request.waitForEvents() + "])",
-                new LocalClusterUpdateTask(request.waitForEvents()) {
-                    @Override
-                    public ClusterTasksResult<LocalClusterUpdateTask> execute(ClusterState currentState) {
-                        return unchanged();
-                    }
+            var updateTask = new LocalClusterUpdateTask(request.waitForEvents()) {
```

Comment:
These `LocalClusterUpdateTask` things are a bit weird, none of them change the cluster state (which seems right and proper) - could we fix their API to reflect that within this change too?

Reply:
This is possible but actually a bit noisy and not really necessary, I will do it in a follow-up instead.